src/main/java/graphql/annotations/DefaultTypeFunction.java

```diff
@@ -59,6 +55,25 @@ void setAnnotationsProcessor(GraphQLAnnotationsProcessor annotationsProcessor) {
         this.annotationsProcessor = annotationsProcessor;
     }
 
+    private class IDFunction implements TypeFunction {
+        @Override
+        public String getTypeName(Class<?> aClass, AnnotatedType annotatedType) {
+            return Scalars.GraphQLID.getName();
+        }
+
+        @Override
+        public boolean canBuildType(Class<?> aClass, AnnotatedType annotatedType) {
+            return annotatedType != null
+                    && (aClass == Integer.class || aClass == int.class || aClass == String.class)
+                    && (annotatedType.getAnnotation(GraphQLID.class) != null);
```

Comment:
You shouldn't use `getAnnotation(GraphQLID.class)` to verify that the annotation is present. It may throw a NullPointerExeception. You should use instead `isAnnotationPresent(GraphQLID.class)`.

Reply:
You're right, thanx :)